src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs

```diff
@@ -170,36 +168,33 @@ public void ItFailsToSetGUISubsystemWithWrongDefault()
         [PlatformSpecific(TestPlatforms.AnyUnix)]
         public void ItGeneratesExecutableImage()
         {
-            using (TestDirectory testDirectory = TestDirectory.Create())
-            {
-                string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
-                string destinationFilePath = Path.Combine(testDirectory.Path, "DestinationAppHost.exe.mock");
-                string appBinaryFilePath = "Test/App/Binary/Path.dll";
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
+            string destinationFilePath = Path.Combine(testDirectory.Path, "DestinationAppHost.exe.mock");
+            string appBinaryFilePath = "Test/App/Binary/Path.dll";
 
-                chmod(sourceAppHostMock, Convert.ToInt32("755", 8)) // match installed permissions: -rwxr-xr-x
-                    .Should()
-                    .NotBe(-1);
+            const UnixFileMode ExecutablePermissions = UnixFileMode.UserRead | UnixFileMode.UserExecute | UnixFileMode.UserWrite |
+                UnixFileMode.GroupRead | UnixFileMode.GroupExecute |
+                UnixFileMode.GroupRead | UnixFileMode.GroupExecute;
 
-                GetLastError()
-                    .Should()
-                    .NotBe(4); // EINTR
-
-                GetFilePermissionValue(sourceAppHostMock)
-                    .Should()
-                    .Be(Convert.ToInt32("755", 8));
+            File.SetUnixFileMode(sourceAppHostMock, ExecutablePermissions);
+            File.GetUnixFileMode(sourceAppHostMock) // match installed permissions: -rwxr-xr-x
+                .Should()
+                .HaveFlag(ExecutablePermissions);
```

Comment:
I believe this will return true even if UnixFileMode.GroupWrite is set as it's using Enum.HasFlag. You want to assert equality.

Reply:
Thanks. Yes with `HaveFlag`, it will only fail when we have fewer permissions, which was the original bug when this test was added so I didn't think about testing excess permission case. Changed to `Be()` for equality check.